### PR TITLE
Feature/305/payment and sign up

### PR DESCRIPTION
**Pull request by antoineeudes**

### CODE REVIEW

File: src/components/organisms/ProfileModal/UserInformationContent/UserInformationContent.tsx [hunk 1]
```diff
@@ -37,9 +39,13 @@ const UserInformationContent: React.FunctionComponent<PropsType> = ({
   const firebase = useFirebase();
   const logout = () => {
     firebase.auth().signOut();
+    // we need to hide the modal because if we already are on the Entrance Page, history.push has no effect
+    hideModal();
     history.push(`/venue/${venue?.id}`);
   };
 
+  if (!user) return <></>;
```
Comment (DotGav) [inline src/components/organisms/ProfileModal/UserInformationContent/UserInformationContent.tsx:47]: Why are you returning an empty div?

File: src/pages/Account/CodeOfConduct.tsx [hunk 2]
```diff
@@ -36,19 +42,20 @@ const QUESTIONS: {
...
+  const { venueId } = getQueryParameters(location.search);
   const { register, handleSubmit, errors, formState, watch } = useForm<
     CodeOfConductFormData
   >({
     mode: "onChange",
   });
   const onSubmit = async (data: CodeOfConductFormData) => {
     await updateUserProfile(user.uid, data);
-    history.push("/");
+    history.push(`/${venueId ? `venue/${venueId}${location.search}` : ""}`);
```
Comment (DotGav) [inline src/pages/Account/CodeOfConduct.tsx:58]: Should an error be displayed if there is no venue or is this to be expected?

File: src/pages/Account/Login.tsx [hunk 3]
```diff
@@ -24,7 +32,7 @@ const Login = () => {
   const onSubmit = async (data: LoginFormData) => {
     try {
       await signIn(data);
-      history.push("/");
+      history.push(`/${venueId ? `venue/${venueId}${location.search}` : ""}`);
```
Comment (DotGav) [inline src/pages/Account/Login.tsx:35]: Same here should the be an error or is it expected to sometimes have no venue?

Comment (antoineeudes) [reply src/pages/Account/Login.tsx:35]: same

File: src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.scss [hunk 4]
```diff
@@ -90,6 +90,12 @@
     .button-container {
       display: flex;
       justify-content: flex-end;
+
+      .buy-tickets-button {
+        width: 150px;
+        height: 50px;
+        font-size: 15px;
```
Comment (BenEllerby) [inline src/components/venues/Jazzbar/EntranceExperience/EntranceExperience.scss:97]: no standard font size?

File: src/utils/getQueryParameters.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const getQueryParameters = (search: string | null) =>
```
Comment (BenEllerby) [inline src/utils/getQueryParameters.ts:1]: Not supported by your router lib?

Comment (DotGav) [reply src/utils/getQueryParameters.ts:1]: I agree should be ```new URLSearchParams(useLocation().search)``` .get("whatever_query_param")

**Pull request merged**